spectator/meter/id.go

```diff
@@ -165,13 +162,36 @@ func (id *Id) WithTags(tags map[string]string) *Id {
 }
 
 func toSpectatorId(name string, tags map[string]string) string {
-	result := invalidCharacters.ReplaceAllString(name, "_")
+	result := replaceInvalidCharacters(name)
 
 	for k, v := range tags {
-		k = invalidCharacters.ReplaceAllString(k, "_")
-		v = invalidCharacters.ReplaceAllString(v, "_")
+		k = replaceInvalidCharacters(k)
+		v = replaceInvalidCharacters(v)
 		result += fmt.Sprintf(",%s=%s", k, v)
 	}
 
 	return result
 }
+
+func replaceInvalidCharacters(input string) string {
+	var result strings.Builder
```

Comment:
I'd recommend use `strings.Map`, which does the same thing you are but with a bunch of arcane optimizations.

Alternatively, if you want to use strings.Builder, consider calling `result.Grow(len(input))` up front so that you only allocate once.